apps/zupoll-client/components/login/Login.tsx

```diff
@@ -18,38 +19,44 @@ import { ZupollError } from "../../src/types";
  */
 export function Login({
   onLogin,
-  onServerLoading,
+  onError,
+  setServerLoading,
   requestedGroup,
   prompt,
   deemphasized,
 }: {
   onLogin: (token: string) => void;
-  onServerLoading: () => void;
+  onError: (error: ZupollError) => void;
+  setServerLoading: (loading: boolean) => void;
   requestedGroup: string;
   prompt: string;
   deemphasized?: boolean;
 }) {
-  const [error, setError] = useState<ZupollError>();
   const [loggingIn, setLoggingIn] = useState(false);
 
   const [pcdStr] = usePassportPopupMessages();
 
+
   useEffect(() => {
     if (!loggingIn) return;
     if (!pcdStr) return;
 
     (async () => {
       try {
-        onServerLoading();
+        setServerLoading(true);
         const token = await fetchLoginToken(requestedGroup, pcdStr);
         onLogin(token);
       } catch (err: any) {
-        setError({ title: "Login failed", message: err.message });
-      } finally {
-        setLoggingIn(false);
+        const loginError = {
+          title: "Login failed",
+          message: err.message,
+        } as ZupollError;
```

Comment:
rather than cast do something like: 
```suggestion
        const loginError: ZupollError = {
          title: "Login failed",
          message: err.message,
        };
```

Reply:
True